File: daily/2026/1727_Largest_Submatrix_With_Rearrangements.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int largestSubmatrix(vector<vector<int>>& matrix) {
        int m = matrix.size();
        int n = matrix[0].size();
        vector<vector<int>> ending(m, vector<int>(n, -1));
        for (int j = 0; j < n; j++){
            int t = m - 1;
            for (int i = m - 1; i >= 0; i--){
                if (matrix[i][j] == 0) t = i - 1;
                else{
                    ending[i][j] = t;
                }
            }
        }
        int ans = 0;
        for (int i = 0; i < m; i++){
            vector<int>& curr = ending[i];
            sort(curr.begin(), curr.end(), greater<int>());
            int temp = m - 1;
            for (int j = 0; j < n; j++){
                if (curr[j] == -1) break;
                temp = min(temp, curr[j]);
                ans = max(ans, (j + 1) * (temp - i + 1));
            }
        }
        return ans;
    }
};

int main(){
    vector<vector<int>> matrix = {{0,0,1},{1,1,1},{1,0,1}};
    // vector<vector<int>> matrix = {{1,0,1,0,1}};
    // vector<vector<int>> matrix = {{1,1,0},{1,0,1}};
    Solution S;
    cout << S.largestSubmatrix(matrix) << endl;
    return 0;
}

File: daily_python/1727_Largest_Submatrix_With_Rearrangements.py
import os
import sys
from typing import List

class Solution:
    def largestSubmatrix(self, matrix: List[List[int]]) -> int:
        m, n = len(matrix), len(matrix[0])
        prev = []
        ans = 0
        for r in range(m):
            curr = []
            visited = [False] * n
            for h, j in prev:
                if matrix[r][j] != 0:
                    curr.append((h+1, j))
                visited[j] = True
            for j in range(n):
                if not visited[j] and matrix[r][j] == 1:
                    curr.append((1, j))
            for l in range(len(curr)):
                ans = max(ans, (l + 1) * curr[l][0])
            prev = curr
        return ans
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            matrix = [list(map(int, row.split(','))) for row in lines[0].split('=')[1].strip()[2:-3].split('},{')]
            dataset.append(matrix)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for matrix in dataset:
        results.append(solution.largestSubmatrix(matrix))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')